transactions.go

```diff
@@ -112,9 +112,135 @@ func (t ItemTransaction) Mutate(b *Battle) {
 		// TODO: remove consumed item from party's inventory
 	}
 	switch t.Item {
-	// ItemCategoryMedicine
-	case ItemPotion:
-		b.QueueTransaction(target.RestoreHP(20))
+	// ItemCategoryHealing
+	case ItemBerryJuice, ItemPotion:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 20,
+		})
+	case ItemEnergyPowder:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 50,
+		})
+		b.QueueTransaction(FriendshipTransaction{
+			Target: target,
+			Amount: [3]int{-5, -5, -10}[target.Friendship/100],
+		})
+	case ItemEnergyRoot:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 200,
+		})
+		b.QueueTransaction(FriendshipTransaction{
+			Target: target,
+			Amount: [3]int{-10, -10, -15}[target.Friendship/100],
+		})
+	case ItemFreshWater:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 50,
+		})
+	case ItemFullRestore:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: target.MaxHP(),
+		})
+		b.QueueTransaction(CureStatusTransaction{
+			Target:       t.Target,
+			StatusEffect: target.StatusEffects,
+		})
+	case ItemHyperPotion:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 200,
+		})
+	case ItemLemonade:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 80,
+		})
+	case ItemMaxPotion:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: target.MaxHP(),
+		})
+	case ItemMoomooMilk:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 100,
+		})
+	case ItemSodaPop:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 60,
+		})
+	case ItemSuperPotion:
+		b.QueueTransaction(HealTransaction{
+			Target: target,
+			Amount: 50,
+		})
```

Comment:
Since a lot of these healing items do the same thing, it would probably be better to put these in a `map` to look up the healing amount.

For `ItemMaxPotion` in particular, you could put `^uint(0)` for the max uint value.